app/models/maestrano/connector/rails/entity.rb

```diff
@@ -11,41 +11,46 @@ class Maestrano::Connector::Rails::Entity < Maestrano::Connector::Rails::EntityB
   # Return an array of entities from the external app
   def get_external_entities(external_entity_name, last_synchronization_date = nil)
     Maestrano::Connector::Rails::ConnectorLogger.log('info', @organization, "Fetching #{Maestrano::Connector::Rails::External.external_name} #{self.class.external_entity_name.pluralize}")
-    # TODO
-    # This method should return only entities that have been updated since the last_synchronization_date
-    # It should also implements an option to do a full synchronization when @opts[:full_sync] == true or when there is no last_synchronization_date
-    # It should also support [:__limit] and [:__skip] opts, meaning that if they are present, it should return only @[:__limit] entities while skipping the @opts[:__skip] firsts
-    # Maestrano::Connector::Rails::ConnectorLogger.log('info', @organization, "Received data: Source=#{Maestrano::Connector::Rails::External.external_name}, Entity=#{self.class.external_entity_name}, Response=#{entities}")
+
+    last_synchronization_date = nil if @opts[:full_sync]
```

Comment:
Here the full synchronization should be triggered if `@opts[:full_sync]` is true or if `last_synchronization_date` is nil 

Something like:
```ruby
if @opts[:full_sync] || !last_synchronization_date
      entities = @external_client.get_entities(external_entity_name, @opts)
    else
      entities = @external_client.get_entities(external_entity_name, @opts, last_synchronization_date)
end
```

Also, does Pipedrive support `limit` and `skip` options? In that case `@opts` should be passed to the client in order to implement the batched call. I.e. Retrieve 25 (limit) records, skipping the first 50( skip).